docs/reference/upgrade/preparing_to_upgrade.asciidoc

```diff
@@ -0,0 +1,23 @@
+[float]
+=== Preparing to upgrade
+
+Before you start to upgrade your cluster to version {version}.
+
+. Check the <<deprecation-logging, deprecation log>> to see if you are using any
+deprecated features and update your code accordingly.
+
+. Review the <<breaking-changes,breaking changes>> and make any necessary
+changes to your code and configuration for version {version}.
+
+. If you use custom plugins, make sure compatible versions are available.
+
+. Test upgrades in a dev environment before upgrading your production cluster.
+
+. <<modules-snapshots,Back up your data by taking a snapshot!>> Once you have
+started to upgrade your cluster to version {version} you must complete the
+upgrade. Rolling a node back to an earlier version is not possible, because
+nodes of version {version} write data in their data paths using a format that
+earlier versions cannot read. The only way to revert an incomplete upgrade is to
+completely delete the contents of the data path on each upgraded node. If this
+results in data loss then you will need to restore the lost data from a
+snapshot.
```

Comment:
I am wondering if this remains open to interpretation. Maybe we should make a stronger statement that suggests you must discard the whole cluster if an upgrade fails.

Reply:
On reflection I think we have to say that. We don't test the node-by-node downgrade I suggested earlier, and it's definitely possible that the cluster state gets changed by a single upgraded node in a way that poisons the older nodes and e.g. prevents a second upgrade attempt.